services/typeddata/hash.go

```diff
@@ -19,6 +19,15 @@ var (
 	int256Type, _  = abi.NewType("int256", nil)
 )
 
+// Hash generates a hash of TypedData and verifies that chainId in the typed data matches currently selected chain.
+func Hash(typed TypedData, chain *big.Int) (common.Hash, error) {
```

Comment:
i would rename to something more verbose cause this method also performs some validation, and hashing usually doesn't

Reply:
@dshulyak do idea, something like `ValidateAndHash`?

Reply:
sounds fine :+1: